v0/destinations/bqstream/transform.js

```diff
@@ -42,4 +51,95 @@ const process = async event => {
   };
 };
 
-module.exports = { process };
+const batchEvents = eventsChunk => {
+  const batchedResponseList = [];
+
+  // arrayChunks = [[e1,e2, ..batchSize], [e1,e2, ..batchSize], ...]
+  const arrayChunks = _.chunk(eventsChunk, MAX_ROWS_PER_REQUEST);
+
+  // list of chunks [ [..], [..] ]
+  arrayChunks.forEach(chunk => {
+    const batchResponseList = [];
+    const metadata = [];
+
+    let batchEventResponse = defaultBatchRequestConfig();
+    const { message, destination } = chunk[0];
+
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      // Pixel code must be added above "batch": [..]
+      batchResponseList.push(ev.message.properties);
+      metadata.push(ev.metadata);
+    });
+
+    batchEventResponse.batchedRequest = {
+      datasetId: message.datasetId,
+      tableId: message.tableId,
+      projectId: message.projectId,
+      properties: batchResponseList
+    };
+
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata,
+      destination
+    };
+
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+};
+
+const processRouterDest = async inputs => {
+  const errorRespEvents = checkInvalidRtTfEvents(inputs, DESTINATION);
+  if (errorRespEvents.length > 0) {
+    return errorRespEvents;
+  }
+
+  const eventsChunk = []; // temporary variable to divide payload into chunks
+  const errorRespList = [];
+  await Promise.all(
```

Comment:
This block looks very similar and I have seen many times, is there util for this? if not please create one.

Reply:
we do not have a util function of this yet and removing async way as mentioned below makes this specific to bqstream.